### ISSUE TITLE
Issue when wrapping eqx.field with additional metadata

### ISSUE THREAD

**Issue by mjo22** [patrick-kidger/equinox#522]
Hi, I need to wrap eqx.field to add additional metadata to my fields, besides just converter and static. When I do this, converter and static are not used at all.

The following toy example produces the unexpected behavior.

```python
def field(*, foo: Optional[bool] = None, **kwargs: Any):
    metadata = kwargs.pop(“metadata”, {})
    if foo is not None:
        metadata[“foo”] = foo
    return eqx.field(metadata=metadata, **kwargs)
```

Now, let’s say I use this wrapper in my Modules, 

```python
module = ExampleModule()
field = dataclasses.fields(module)[0]
print(field.metadata)
```

where, say, I have written this example field with something like `bar: Array = field(foo=True, default=1.0, converter=jnp.asarray)`. The converter will not appear in the metadata in the above example, and indeed

```python
module = ExampleModule()
assert isinstance(module.bar, Array)
```

fails. Is this by design, or is this a bug? Thanks so much!

**Comment by patrick-kidger** [comment]
Thanks for the report! This is not by design, and should be fixed in #530.

**Comment by patrick-kidger** [comment]
Just merged #530 into the `dev` branch.
This will be available in the upcoming v0.11.0 release later this week.
